vic/CMakeLists.txt

```diff
@@ -0,0 +1,207 @@
+            /lib/
+            /lib/openmpi
+            /lib64/
+            /usr/lib
+            /usr/lib/openmpi
+            /usr/lib/x86_64-linux-gnu
+            /usr/lib64
+            /usr/local/lib
+            /usr/local/lib64
+            /usr/local/openmpi/lib
+       )
+    set(MPI_INCLUDE_SEARCH_PATHS
+            /usr/include
+            /usr/local/include
+       )
+    set(MPICH_INCLUDE_SEARCH_PATHS
+            $ENV{HOME}/mpich/include
+            /usr/include/mpich
+            /usr/local/include/mpich
+            /usr/local/mpich/include
+       )
+    set(OPENMPI_INCLUDE_SEARCH_PATHS
+            $ENV{HOME}/openmpi/include
+            /usr/include/openmpi
+            /usr/local/include/openmpi
+            /usr/local/openmpi/include
+       )
+
+    # Find MPI library path.
+    find_path(MPI_LIB_DIR NAMES libmpich.so PATHS ${MPI_LIB_SEARCH_PATHS})
+
+    if(NOT MPI_LIB_DIR)
+        find_path(MPI_LIB_DIR NAMES libmpi.so PATHS ${MPI_LIB_SEARCH_PATHS})
+        if(NOT MPI_LIB_DIR)
+            message(WARNING "Could not find MPI library. VIC Image Driver would not be built.")
+            set(BUILD_IMAGE OFF)
+        else()
+            message(STATUS "OpenMPI library found, at ${MPI_LIB_DIR}")
+            set(BUILD_IMAGE ON)
+            link_directories(${MPI_LIB_DIR})
+            set(MPI_LIB mpi)
+            set(MPI_INCLUDE_SEARCH_PATHS ${MPI_INCLUDE_SEARCH_PATHS} ${OPENMPI_INCLUDE_SEARCH_PATHS})
+        endif()
+    else()
+        message(STATUS "MPICH library found, at ${MPI_LIB_DIR}")
+        set(BUILD_IMAGE ON)
+        link_directories(${MPI_LIB_DIR})
+        set(MPI_LIB mpich)
+        set(MPI_INCLUDE_SEARCH_PATHS ${MPI_INCLUDE_SEARCH_PATHS} ${MPICH_INCLUDE_SEARCH_PATHS})
+    endif()
+
+    # Find MPI include path.
+    find_path(MPI_INCLUDE_DIR NAMES mpi.h PATHS ${MPI_INCLUDE_SEARCH_PATHS})
+
+    if(NOT MPI_INCLUDE_DIR)
+        message(WARNING "Could not find MPI include. VIC Image Driver would not be built.")
+        set(BUILD_IMAGE OFF)
+    else()
+        message(STATUS "MPI include found, at ${MPI_INCLUDE_DIR}")
+        set(BUILD_IMAGE ON)
+        include_directories(${MPI_INCLUDE_DIR})
+    endif()
+endif()
+
+#           Find out the include and library directory of netCDF.
+if(IMAGE)
+
+    # Probably include and library paths of netCDF.
+    set(NETCDF_INCLUDE_SEARCH_PATHS
+            /usr/include
+            /usr/include/netcdf
+            /usr/local/include
+            /usr/local/include/netcdf
+            /usr/local/netcdf/include)
+    set(NETCDF_LIB_SEARCH_PATHS
+            /lib/
+            /lib64/
+            /usr/lib
+            /usr/lib/netcdf
+            /usr/lib/x86_64-linux-gnu
+            /usr/lib64
+            /usr/local/lib
+            /usr/local/lib64
+            /usr/local/netcdf/lib)
+
+    find_path(NETCDF_INCLUDE_DIR NAMES netcdf.h PATHS ${NETCDF_INCLUDE_SEARCH_PATHS})
+    find_path(NETCDF_LIB_DIR NAMES libnetcdf.so PATHS ${NETCDF_LIB_SEARCH_PATHS})
```

Comment:
Remove hard coded paths.  Take a look at the existing image driver makefile for an example on how to do this.  `nc-config -- libs` and `nc-config --cflags` should give you the necessary information.  Users should be able to overwrite those though.